server/src/main/java/org/elasticsearch/usage/UsageService.java

```diff
@@ -89,6 +92,33 @@ public void addRestHandler(BaseRestHandler handler) {
         }
     }
 
+    public static final String OTHER_SUBTYPE = "other";
+
+    public void registerAggregationUsage(String aggregationName) {
+        registerAggregationUsage(aggregationName, OTHER_SUBTYPE);
+    }
+
+    public void registerAggregationUsage(String aggregationName, String valuesSourceType) {
+        Map<String, LongAdder> subAgg = aggs.computeIfAbsent(aggregationName, k -> new HashMap<>());
+        if ( subAgg.put(valuesSourceType, new LongAdder()) != null) {
+            throw new IllegalArgumentException("stats for aggregation [" + aggregationName + "][" + valuesSourceType +
+                "] already registered");
+        }
+    }
+
+    public void incAggregationUsage(String aggregationName, String valuesSourceType) {
+        Map<String, LongAdder> valuesSourceMap = aggs.get(aggregationName);
+        // Not all aggs register their usage at the moment we also don't register them in test context
```

Comment:
Are there other aggs that aren't caught by the `else` in SearchModule (using `registerAggregationUsage(String aggregationName) `)?

Reply:
Hmm, good point. Let's find out.